plugin/src/main/kotlin/com/automattic/android/PublishToS3Plugin.kt

```diff
@@ -2,12 +2,78 @@ package com.automattic.android.publish
 
 import com.automattic.android.publish.CalculateVersionNameTask
 import com.automattic.android.publish.CheckS3VersionTask
+import com.automattic.android.publish.PublishToS3PluginExtension
 import org.gradle.api.Project
 import org.gradle.api.Plugin
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.plugins.ExtraPropertiesExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.credentials.AwsCredentials
+import org.gradle.api.publish.maven.tasks.PublishToMavenRepository
+import java.net.URI
 
-class PublishToS3Plugin: Plugin<Project> {
+const val EXTRA_VERSION_NAME = "extra_version_name"
+
+class PublishToS3Plugin : Plugin<Project> {
     override fun apply(project: Project) {
+        val extension = project.extensions.create("configureS3PublishPlugin", PublishToS3PluginExtension::class.java)
+        val extraProperties = project.getExtensions().getByType(ExtraPropertiesExtension::class.java)
+        extraProperties.set("s3PublishVersion", "testingExtraProperties")
+
+        project.plugins.apply("maven-publish")
+
         project.tasks.register("calculateVersionName", CalculateVersionNameTask::class.java)
-        project.tasks.register("isVersionPublishedToS3", CheckS3VersionTask::class.java)
+        project.tasks.register("isVersionPublishedToS3", CheckS3VersionTask::class.java) {
```

Comment:
We probably want a `shouldPublishToS3` task so we can use it before building the project.